### DISCUSSION TITLE
Options for max width/height or keep aspect ratio for Camera

### DISCUSSION THREAD

**Discussion by zyniq82** [ionic-team/capacitor#1952] [Ideas]
I have a feature request about adding options to the Camera plugin for setting max `width`/`height`, alternatively a new option for keeping aspect ratio (e.g. `keepRatio`).

This is somewhat related to #217, in which @mlynch wrote:
> _Just added this feature to master and will release tomorrow. you can set `width` and/or `height`. If you set one but not the other, it does an aspect ratio scale to fit that dimension._

What I need is a further extension of the scaling capabilities, I want to be able to limit the _longest_ side of the photo, yet keep the aspect ratio. I've used the [Cordova Camera plugin](https://cordova.apache.org/docs/en/latest/reference/cordova-plugin-camera/index.html) before, in which this was possible.

The problem now is that
- if I specify both `width` _and_ `height`, I'm sure no side is too long, but then the aspect ratio of the resulting image is changed.
- if I specify only `width` _or_ `height`, the aspect ratio is retained, but then I don't know if my prescribed value is applied to the longer or shorter side of the image, since I don't know beforehand if the image is in portrait or landscape format.

I don't know if it's good that `width` and `height` work like in the Cordova case, always retaining aspect ratio, or if it's better to have an additional option enforcing that aspect ratio is kept (e.g. `keepRatio`). The latter alternative gives more flexibility, but maybe makes things too complicated.

Let me know what you think of this!
Is this a feature that can be added?

**Comment by andrescera** [comment]
I've got almost the same problem but, for me none o those options (setting a value for both width and height or setting just one) return an image which keeps the aspect ration in opposition what it does is creating an image with a "Compressed ratio", what i mean is that it keeps it resizes the picture without cropping it or maintaining its aspect ration 

Original/After pressing OK
![5ffb3372-9037-44b1-8637-f38ec2c53505](https://user-images.githubusercontent.com/20803123/64830808-170d3e80-d598-11e9-83ed-05679a9f4721.jpeg)
![938b598a-ce62-4082-b910-85893a22808d](https://user-images.githubusercontent.com/20803123/64830809-170d3e80-d598-11e9-9e0f-c7bd9faac50a.jpeg)

Affected platform
- [x] Android
- [x] iOS
- [ ] electron
- [ ] web

OS of the development machine
- [ ] Windows
- [x] macOS
- [ ] linux

Capacitor version: 1.2.0

**Comment by zyniq82** [comment]
For reference, it seems that the problem @Cclown98 is experiencing is not the problem for which I've opened this issue. However I think @Cclown98's issue may be fixed by pull request #2123 (correct me if I'm wrong).

**Comment by imhoffd** [comment]
This will be addressed in this feature request: https://github.com/ionic-team/capacitor-plugins/issues/2
